assimilation_code/modules/io/direct_netcdf_mod.f90

```diff
@@ -1556,12 +1572,14 @@ subroutine write_variables(ncid, var_block, start_var, end_var, domain, &
       dims = get_io_dim_lengths(domain, i)
 !>@todo FIXME, the first variable in the second domain is not found when using coamps_nest.
       ret = nf90_inq_varid(ncid, trim(get_variable_name(domain, i)), var_id)
-      call nc_check(ret, 'write_variables:', 'nf90_inq_varid "'//trim(get_variable_name(domain,i))//'"')
+      write(msgstring,*) 'nf90_inq_varid "'//trim(get_variable_name(domain,i))//'"'
+      call nc_check(ret, 'write_variables:', msgstring)
 
-      if (missing_possible) call set_model_missing_value(var_block(istart:iend), domain, i)
+      if (allow_missing) call set_model_missing_value(domain, i, var_block(istart:iend))
```

Comment:
this is only changing the order of the arguments to this routine?  that is a very error-prone change and unless there is a good reason i'm not a fan.  this is an internal-only, non-public routine so you can should be able to verify that all the calls have been correctly changed, but in most situations i'd downvote an argument-order shuffle.